### PR TITLE
integration test on databricks platform

### PR DESCRIPTION
**Pull request by pawel-big-lebowski**
### Problem

Spark integration misses integration test on Databricks. This is important as many community users come with databricks scenarios and behaviour on databricks platform may differ from one on locally run vanilla-spark. 

Closes: #1882

### Solution

 * Prepare a test that spawns spark session on databricks cluster. 
 * Include it within CircleCI with the ability to being run manually.
 * Running databricks integration test requires extra parameters param: `databricks-test` set to `enabled`

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [ ] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [ ] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

**Comment by pawel-big-lebowski** [comment]
Succeeded databricks integration test when triggered manually:
https://app.circleci.com/pipelines/github/OpenLineage/OpenLineage/7402/workflows/caddc283-880c-459d-b318-dd6a3f26b67c/jobs/120447

```
DatabricksIntegrationTest > testCreateTableAsSelect() PASSED
```

**Comment by pawel-big-lebowski** [comment]
@mobuchowski @JDarDagran I've included databricks cluster logs: https://app.circleci.com/pipelines/github/OpenLineage/OpenLineage/7430/workflows/da5fe2e9-20f1-44c0-879e-9baf116898d7/jobs/121073

File: .circleci/continue_config.yml [hunk 1]
```diff
@@ -453,18 +458,64 @@ jobs:
...
+            - attach_workspace:
+                at: ~/.m2/repository/io/openlineage/
+            - run: |
+                sudo update-alternatives --set java /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java
+                sudo update-alternatives --set javac /usr/lib/jvm/java-8-openjdk-amd64/bin/javac
+            - run: ./gradlew shadowJar -x test
+            - run: ./gradlew --no-daemon databricksIntegrationTest -x test   -Pspark.version=<< parameters.spark-version >> -PdatabricksHost=$DATABRICKS_HOST -PdatabricksToken=$DATABRICKS_TOKEN
+            - run:
+                when: on_fail
+                command: cat app/build/test-results/databricksIntegrationTest/TEST-*.xml
```
Comment (JDarDagran) [inline .circleci/continue_config.yml:503]: Is this neccessary? There is `store_artifacts` step below that should contain the same and is more convenient to browse.

Comment (pawel-big-lebowski) [reply .circleci/continue_config.yml:503]: It's been added within spark integration tests as well and I wanted to be consistent with that. 
Assuming this was failing for over a year and no-one has noticed that, I don't think it's necessary. at all

**Review by mobuchowski — APPROVED** [review]

**Pull request merged**